onecodex/lib/inline_validator.py

```diff
@@ -249,7 +251,15 @@ def __iter__(self):
             while True:
                 match = self.seq_reader.match(self.unchecked_buffer, end)
                 if match is None:
+                    if (waiting_for_data or eof) and len(self.unchecked_buffer) - end > 0:
+                        raise ValidationError('Sequence in buffer does not match expected format.')
```

Comment:
Let's have this say "Your FASTA/Q file terminates abruptly or is otherwise malformed"